### PR TITLE
N-Hentai Source

### PR DESCRIPTION
**Pull request by ConradWeiser**
## N-Hentai Source

Basic features:
* Source to incorperate [nHentai](nhentai.net/) into the Application
* Supports searching for 6-digit titles using a switching search case
* Homepage populated with the front-page of the website (as you'd expect from a source)

Notes:
* Everything is hard-coded tagged as hentai, given what the source is
* Manga update searching does not exist, since nHentai doesn't have chapters

[nice](https://external-content.duckduckgo.com/iu/?u=http%3A%2F%2Fi2.kym-cdn.com%2Fphotos%2Fimages%2Ffacebook%2F000%2F764%2F965%2F47a.jpg)

### CODE REVIEW

**Comment by ConradWeiser** [comment]
Note that the checks are failing due to a issue which needs to be fixed at #5

File: src/sources/NHentai/NHentai.ts [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+export class NHentai extends Source {
+  constructor(cheerio: CheerioAPI) {
+    super(cheerio)
+  }
+
+  get version(): string { return 'v0.5' }
+  get name(): string { return 'nHentai' }
+  get description(): string { return 'Extension that pulls manga from nHentai' }
+  get author(): string {return 'Conrad Weiser'}
+  get icon(): string {return "https://i.imgur.com/uLAimaY.png"} // The website has SVG versions, I had to find one off of a different source
```
Comment (DanielKovalevich) [inline src/sources/NHentai/NHentai.ts:23]: Icon will need to be a local directory reference. This was changed in the last update of the code.
This icon will reside inside a folder called `includes` within the extension folder.
Refer to the other extensions for an example

File: src/sources/NHentai/NHentai.ts [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+import { APIWrapper } from '../../API'
+
+const NHENTAI_DOMAIN = 'https://nhentai.net'
+
+export class NHentai extends Source {
+  constructor(cheerio: CheerioAPI) {
+    super(cheerio)
+  }
+
+  get version(): string { return 'v0.5' }
```
Comment (DanielKovalevich) [inline src/sources/NHentai/NHentai.ts:19]: To remain consistent with other extensions, remove the `v` in front of the version number

File: src/sources/NHentai/NHentai.ts [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    return createRequestObject({
+      url: `http://niceme.me`,              // We do not use this method, but we need to implement it. Load a page which is tiny in size I guess
+      method: "GET",
+    })
+  }
+
+  filterUpdatedManga(data: any, metadata: any): { 'updatedMangaIds': string[], 'nextPage': boolean } {
+    // There's no chapters, so we can just ignore empty stuff! Horray!
+    return {updatedMangaIds : [], nextPage: false}
+  }
```
Comment (DanielKovalevich) [inline src/sources/NHentai/NHentai.ts:227]: These functions do not need to be implemented as they return null by default in the parent Source class

File: src/sources/NHentai/NHentai.ts [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+    }
+
+    section[0].items = updatedHentai
+    return section
+  }
+
+  getViewMoreRequest(key: string, page: number): Request | null { return null }
+  getViewMoreItems(data: any, key: string): MangaTile[] | null { return null }
+  getTagsRequest(): Request | null { return null }
+  getTags(data: any): TagSection[] | null { return null }           // Temporarily disabled, there are 26 pages of tags. I'd like to discuss with paper how to handle this
```
Comment (DanielKovalevich) [inline src/sources/NHentai/NHentai.ts:354]: Implement the functions and return the Tags. The app will dynamically handle different tags by the different sources.

File: src/sources/NHentai/NHentai.ts [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+            image: image
+        })
+    }
+
+    section[0].items = updatedHentai
+    return section
+  }
+
+  getViewMoreRequest(key: string, page: number): Request | null { return null }
+  getViewMoreItems(data: any, key: string): MangaTile[] | null { return null }
```
Comment (DanielKovalevich) [inline src/sources/NHentai/NHentai.ts:352]: If you do not plan to implement these, then they can be removed. They are already defined in Source.ts

File: src/sources/NHentai/NHentai.ts (hunk 4 above)
Comment (DanielKovalevich) [reply src/sources/NHentai/NHentai.ts:354]: Reread comment. Just grab the most popular tags

Comment (ConradWeiser) [reply src/sources/NHentai/NHentai.ts:354]: Will implement tagging in a future version

File: src/sources/NHentai/NHentai.ts (hunk 5 above)
Comment (ConradWeiser) [reply src/sources/NHentai/NHentai.ts:352]: Will be implemented in a future version, references removed for now

**Pull request merged**